packages/libs/eda/src/lib/core/utils/data-element-constraints.ts

```diff
@@ -420,10 +430,16 @@ export function mergeMaxNumValues(
  */
 export function ancestorEntitiesForEntityId(
   entityId: string,
-  entities: StudyEntity[]
+  entities: StudyEntity[],
+  childFilter: (entity: StudyEntity) => boolean = () => true
 ): StudyEntity[] {
   const ancestors = entities.reduceRight((ancestors, entity) => {
```

Comment:
It looks like this relies on `entities` being in a particular order. That makes me nervous about potential future changes. I wonder if it would be better for this function to take a root entity, and perform a tree traversal to find ancestor entities; but this might require upstream changes, so might be better to add a `TODO` comment for now?

Reply:
Indeed - you've commented almost exactly this before. I think I added some comments that time.
Is it time to dive in and fix properly?